### PR TITLE
Substantially broaden what can appear as the argument to FOR

### PR DESCRIPTION
**Pull request by msullivan**
Introduce a notion of AtomicExpr and allow it as the argument to FOR.

### CODE REVIEW

File: edb/edgeql/parser/parser.py [hunk 1]
```diff
@@ -116,8 +117,14 @@ def get_exception(self, native_err, context, token=None):
                     else:
                         msg = f'Unexpected {token.text()!r}'
                 elif rule == 'for iterator':
-                    msg = (f"missing '{{' before {token.text()!r} "
-                           f"in a FOR query")
+                    msg = ("Complex expressions in a FOR query must be "
```
Comment (elprans) [inline edb/edgeql/parser/parser.py:120]:
```suggestion
                    msg = ("Complex expressions in a FOR iterator clause must be "
```

Comment (elprans) [reply edb/edgeql/parser/parser.py:120]: Or, maybe "missing parentheses around complex expression in a FOR iterator clause"

Comment (msullivan) [reply edb/edgeql/parser/parser.py:120]: If we do the latter suggestion, should we also change "Statement used as an expression must be enclosed in a set of parentheses" to "Missing parentheses around statement used as expression"?

Comment (elprans) [reply edb/edgeql/parser/parser.py]: Sure

**Review by elprans — APPROVED** [review]

File: edb/edgeql/parser/parser.py (hunk 1 above)
Comment (msullivan) [reply edb/edgeql/parser/parser.py]: Hm, I think I prefer the first approach, actually. Fine to stick with that?

**Pull request merged**